experimental/DoubleComplex/docs/src/user_interface.md

```diff
@@ -0,0 +1,90 @@
+```@meta
+CurrentModule = Oscar
+```
+# Double complexes -- the user's interface
+We briefly review the mathematical notion of a double complex. 
+Let ``\mathcal A`` be an Abelian category. A double complex 
+``D_{\bullet, \bullet}`` consists of a collection of objects ``D_{i, j}`` in 
+``\mathcal A`` with indices ``(i, j) \in \mathbb Z^2`` and usually arranged 
+in a matrix-like grid, together with two collections 
+of morphisms ``D_{i, j} \to D_{i \pm 1, j}``, the *horizontal* morphisms, and 
+``D_{i, j} \to D_{i, j \pm 1}``, the *vertical* morphisms, so that both 
+the rows and the columns of ``D_{\bullet, \bullet}`` are complexes in the 
+classical sense and such that all resulting squares of maps commute.
+
+In practice one usually encounters complexes which are *bounded* in the sense 
+that outside some specified area of indices ``(i, j) \in \mathbb Z^2`` the entries 
+``D_{i, j}`` are all zero. Such entries are then usually omitted. 
+
+## Basic getters and attributes
+In OSCAR the generic functionality for double complexes is declared for the 
+abstract type `AbsDoubleComplexOfMorphisms`. These functions comprise
+```julia
+  getindex(D::AbsDoubleComplexOfMorphisms, i::Int, j::Int) # Get the `(i,j)`-th entry of `D`
```

Comment:
How about a function to test if a specific index is available, i.e. something like `has_index(D, i, j)` to test whether an object at that position is already computed.

And then separately something like `can_compute_index(D, i, j)` which returns `true` if either the object `D[i,j]` already is there, or if it is not there but could be computed.